bazel/grpc.patch

```diff
@@ -24,16 +24,15 @@ index 1bb970e049..81265483e9 100644
      ],
  )
  
-diff --git a/src/core/lib/channel/channel_args.h b/src/core/lib/channel/channel_args.h
-index 38bb070213..b53086e680 100644
---- a/src/core/lib/channel/channel_args.h
-+++ b/src/core/lib/channel/channel_args.h
-@@ -284,7 +284,7 @@ class ChannelArgs {
+diff --git a/src/core/lib/promise/detail/promise_like.h b/src/core/lib/promise/detail/promise_like.h
+--- a/src/core/lib/promise/detail/promise_like.h	2024-04-18 19:47:19.524155042 +0000
++++ b/src/core/lib/promise/detail/promise_like.h	2024-04-18 23:40:29.050761807 +0000
+@@ -71,7 +71,7 @@
+ 
+ template <typename F>
+ class PromiseLike<F, absl::enable_if_t<!std::is_void<
+-                         typename std::result_of<F()>::type>::value>> {
```

Comment:
@yanavlasov how long are these patches going to live?

Reply:
Unfortunately until gRPC supports C++20. I have inquired about their plans to transition to C++20 but do not have a commitment date at this point.

Reply:
More specifically when gRPC lifts minimum supported C++ version to 17. They are now on C++ 14. There are parts of C++ that do not overlap between C++20 and C++14. This patch fixes one of them.

@markdroth mentioned it will be years before this changes.

Reply:
I believe this particular issue was addressed in https://github.com/grpc/grpc/pull/36016.

For the broader topic of changes related to C++20, @veblush may be able to advise.

Reply:
Excellent. What a time saver!

Reply:
@yanavlasov do you plan to update this PR?

Reply:
If you mean update the PR to avoid this patch, then no, as this C++20 fix has not yet made into a stable release.  It will go away at the next gRPC upgrade.